fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -231,29 +237,47 @@ private boolean insertField(LuceneDocumentFromRecord.DocumentField field, final
                 throw new RecordCoreArgumentException("Invalid type for lucene index field", "type", field.getType());
         }
         document.add(luceneField);
-        return suggestionAdded;
+    }
+
+    @Nonnull
+    private String getStringValue(@Nonnull LuceneDocumentFromRecord.DocumentField field) {
+        switch (field.getType()) {
+            case TEXT:
+            case STRING:
+                return (String) field.getValue();
+            case INT:
+            case LONG:
+            case DOUBLE:
+                return String.valueOf(field.getValue());
+            case BOOLEAN:
+                return ((Boolean) field.getValue()).toString();
+            default:
+                throw new RecordCoreArgumentException("Invalid type for lucene index field", "type", field.getType());
+        }
     }
 
     private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
                                byte[] primaryKey) throws IOException {
-        final IndexWriter newWriter = getOrCreateIndexWriter(state, indexAnalyzer, executor, groupingKey);
+        final TextLanguage language = TextLanguage.getLanguageForTexts(fields.stream().map(this::getStringValue).collect(Collectors.toList()));
```

Comment:
Would it make more sense to only do the full-text fields? It seems odd to count digits from numbers or the letter of "true" and "false" toward the non-CJK counts.

Reply:
Make sense. Using a filter to only check the ```TEXT``` fields now.